### PR TITLE
Adding a Makefile

### PR DESCRIPTION
**Pull request by nikhiljindal**
Adding a Makefile with standard rules.

Makefile inspired from https://github.com/kubernetes/ingress-nginx/blob/b68feb9c590e2c11bdb098549f3208f80029daf7/Makefile

cc @csbell @G-Harmon @madhusudancs @perotinus

### CODE REVIEW

**Comment by nikhiljindal** [comment]
Found out that go fmt, lint and vet are failing.
Filed #59, #60 and #61.

Also found out that our current unit test coverage is ~50-60%. (least is 39% and max is 90%). Details: https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/issues/48#issuecomment-343065547

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2017 The Kubernetes Authors.
```
Comment (perotinus) [inline Makefile]: Shouldn't the copyright be for Google?

Comment (nikhiljindal) [reply Makefile]: aah great catch! Copy paste error :)
fixed.

**Comment by nikhiljindal** [comment]
Thanks @perotinus 
Updated as per comments

**Comment by nikhiljindal** [comment]
Merging so that `/test all` starts to work once https://github.com/kubernetes/test-infra/pull/5373 is merged.

**Pull request merged**